ruby/beard_checker.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class BeardChecker
```

Comment:
I'm thinking, could the class be a Person? and the attributes are either for the Beard, or for their person.

bob = Person
bob.facial_hair_length = 5
bob.has_a_beard?

Reply:
the idea is that it maps to concept in the legislation. Person is a concept in the corpus of law.

Reply:
I'm thinking of it as a service object - `Person` also relates to the reasons for beard and being in Australia 🤔

Reply:
and all sorts of other things - you wouldn't want to have to define a person's beard length when you're using them for something else

Reply:
do you want to do it the way you're thinking of in Python @Br3nda and we can compare approaches and discuss pros and cons

Reply:
yup, if you were wanting bob.owned_income_tax, it would either be nil, or 0 or raise an exception unless you've defined enough to calculate it.

The legislation interacts with each other, they're seldom self contained. Clauses in Social Security refer to Income Tax, and vice versa. But they share a concept of person

Reply:
Maybe we do examples both ways.

Reply:
So the way I've got it now is that it's something you would refer to in the Person class - eg

```
class Person

def initialize(facial_hair: MuttonChops.new)
  @facial_hair = facial_hair
end

def has_beard?
  @facial_hair.is_beard?
end

end
```

```
class MuttonChops << FacialHair
  def initialize(length: 5, color: 'brown', friendly: false)
     @length = length
     @color = color
     @friendly = friendly
  end

  def is_beard?
    BeardChecker.is_beard?(length_in_mm: @length, on_or_below_chin: false, unbroken_between_ears: @friendly)
  end
end
```